### PR TITLE
fix: only return word with desirable alpha characters

### LINKED ISSUE #113: `random.word()` can return undesirable non-alpha characters
### Describe the bug

The pool of strings from which `random.words()` pulls values includes strings that have parentheses, e.g. `Cocos (Keeling) Islands` and `Killer Whale (Orca)`. This can result in `(Keeling)` and `(Orca)` being returned.

There's a larger problem here of non-alphabetic characters being returned in words. Obviously some non-alpha characters are desired in words, e.g. `-`, but some are not. We should decide on the set of acceptable characters that `word()` can return and regex the others out.

### Reproduction

Mock `faker.fake` with the above mentioned values and you can see the bad returns.

### Additional Info

### PR DESCRIPTION
**Pull request by Shinigami92**
fix #113

### CODE REVIEW

File: src/random.ts [hunk 1]
```diff
@@ -286,8 +285,11 @@ export class Random {
     ];
 
     // randomly pick from the many faker methods that can generate words
-    const randomWordMethod = this.faker.random.arrayElement(wordMethods);
-    const result = this.faker.fake('{{' + randomWordMethod + '}}');
+    let result: string;
+    do {
+      const randomWordMethod = this.faker.random.arrayElement(wordMethods);
+      result = this.faker.fake('{{' + randomWordMethod + '}}');
+    } while (!/^[\w\s]+$/.test(result));
```
Comment (ST-DDT) [inline src/random.ts:292]: Does this method work for non latin letter languages?

Comment (Shinigami92) [reply src/random.ts:292]: Good question: I will research this...

Comment (Shinigami92) [reply src/random.ts]: Please have a new look. I know we cannot handle every special case this way, but currently I do not see a better solution due to we do not have `\p` in js regex 🙁 
So I think for now this is enough to check and if there is any issue in the future, we can patch it

File: src/random.ts [hunk 2]
```diff
@@ -285,9 +284,37 @@ export class Random {
...
+      '_',
+      '-',
+    ];
+    let result: string;
+    do {
+      // randomly pick from the many faker methods that can generate words
+      const randomWordMethod = this.faker.random.arrayElement(wordMethods);
+      result = this.faker.fake('{{' + randomWordMethod + '}}');
+    } while (bannedChars.some((char) => result.includes(char)));
     return this.faker.random.arrayElement(result.split(' '));
```
Comment (ST-DDT) [inline src/random.ts:318]: Since space is banned there is no longer a need to select here.

Comment (Shinigami92) [reply src/random.ts:318]: Oh damn, I implemented it wrong! There are some strings that contains spaces so I need to allow it back.

**Review by prisis — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**